### PR TITLE
expose boring ssl version info in the system.build_options table

### PR DESCRIPTION
**Pull request by bharatnc**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Expose BoringSSL version related info in the `system.build_options` table.

ClickHouse uses BoringSSL which is a fork of OpenSSL. It would be nice to have the `OPENSSL_VERSION` used by the ClickHouse in the `system.build_options` table. Also might be somewhat intuitive to have `OPENSSL_IS_BORING_SSL` flag exposed as well:

```
SELECT *
FROM system.build_options
WHERE (name = 'OPENSSL_IS_BORING_SSL') OR (name = 'OPENSSL_VERSION')
┌─name──────────────────┬─value──┐
│ OPENSSL_VERSION       │ 1.1.0g │
│ OPENSSL_IS_BORING_SSL │ 1      │
└───────────────────────┴────────┘
```

### CODE REVIEW

File: contrib/boringssl-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -705,3 +705,109 @@ target_compile_options(_crypto PRIVATE -Wno-gnu-anonymous-struct)
...
+
+  set(${DEC} ${_res} PARENT_SCOPE)
+endfunction()
+
+# ClickHouse uses BoringSSL which is a fork of OpenSSL.
+# This populates CMAKE var OPENSSL_VERSION from the OPENSSL_VERSION_NUMBER defined
+# in contrib/boringssl/include/openssl/base.h. It also sets the CMAKE var OPENSSL_IS_BORING_SSL
+# if it's defined in the file. Both OPENSSL_VERSION and OPENSSL_IS_BORING_SSL variables will be
+# used to populate flags in the `system.build_options` table for more context on ssl version used.
+# This cmake script is adopted from FindOpenSSL cmake module and slightly modified for this use-case .
```
Comment (bharatnc) [inline contrib/boringssl-cmake/CMakeLists.txt:746]: https://github.com/Kitware/CMake/blob/master/Modules/FindOpenSSL.cmake#L519

**Review by yakov-olkhovskiy — APPROVED** [review]

**Comment by bharatnc** [comment]
There are a few failed tests but they don't look related to me.

**Comment by yakov-olkhovskiy** [comment]
@bharatnc me too, I think it's safe to merge

**Comment by bharatnc** [comment]
> @bharatnc me too, I think it's safe to merge

Sure, @yakov-olkhovskiy. In that case can you merge? I don't have the permissions.

**Pull request merged**